cpp/src/examples/babyllama/baby_llama_handler.cc

```diff
@@ -0,0 +1,269 @@
+    }
+  }
+
+  return batch_ivalue;
+}
+
+torch::Tensor LlmHandler::Inference(
+    std::shared_ptr<torch::jit::script::Module> model,
+    std::vector<torch::jit::IValue>& inputs,
+    std::shared_ptr<torch::Device>& device,
+    std::pair<std::string&, std::map<uint8_t, std::string>&>& idx_to_req_id,
+    std::shared_ptr<torchserve::InferenceResponseBatch>& response_batch) {
+  std::vector<torch::Tensor> tensor_vector;
+  auto tokens_list_tensor = inputs[0].toTensor();
+
+  int64_t num_elements = tokens_list_tensor.numel();
+
+  int steps = 256;
+  // // Convert the tensor to a vector of long values
+  std::vector<long> long_vector;
+  long_vector.reserve(num_elements);
+
+  auto data_ptr = tokens_list_tensor.data_ptr<int64_t>();
+  for (int64_t i = 0; i < num_elements; ++i) {
+    long_vector.push_back(data_ptr[i]);
+  }
+
+  int* prompt_tokens = new int[num_elements];
+  for (int64_t i = 0; i < num_elements; ++i) {
+    prompt_tokens[i] = static_cast<int>(long_vector[i]);
+  }
+
+  // start the main loop
+  long start =
+      0;     // used to time our code, only initialized after first iteration
+  int next;  // will store the next token in the sequence
+  int token = prompt_tokens[0];  // kick off with the first token in the prompt
+  int pos = 0;                   // position in the sequence
+  while (pos < steps) {
+    // forward the transformer to get logits for the next token
+    float* logits = forward(&transformer, token, pos);
+
+    // advance the state state machine
+    if (pos < num_elements - 1) {
+      // if we are still processing the input prompt, force the next prompt
+      // token
+      next = prompt_tokens[pos + 1];
+    } else {
+      // otherwise sample the next token from the logits
+      next = sample(&sampler, logits);
+    }
+    pos++;
+
+    torch::Tensor tensor = torch::tensor(next, torch::kLong);
+    tensor_vector.push_back(tensor);
+
+    // data-dependent terminating condition: the BOS (=1) token delimits
+    // sequences
+    if (next == 1) {
+      break;
+    }
+
+    token = next;
+
+    // init the timer here because the first iteration can be slower
+    if (start == 0) {
+      start = time_in_ms();
+    }
+  }
+
+  // report achieved tok/s (pos-1 because the timer starts after first
+  // iteration)
+  if (pos > 1) {
+    long end = time_in_ms();
+    auto token_per_sec = (pos - 1) / (double)(end - start) * 1000;
+    std::cout << "Achieved tok per sec: " << token_per_sec << std::endl;
+  }
+
+  delete[] prompt_tokens;
+
+  torch::Tensor stacked_tensor = torch::stack(tensor_vector);
+  return stacked_tensor;
+}
+
+void LlmHandler::Postprocess(
+    const torch::Tensor& data,
+    std::pair<std::string&, std::map<uint8_t, std::string>&>& idx_to_req_id,
+    std::shared_ptr<torchserve::InferenceResponseBatch>& response_batch) {
+  for (const auto& kv : idx_to_req_id.second) {
+    try {
+      int64_t num_elements = data.numel();
```

Comment:
Can you enable batch processing here as well?

Reply:
Working on batch processing. will keep you posted once it is done